draft-ietf-oauth-attestation-based-client-auth.md

```diff
@@ -477,26 +514,48 @@ Implementers should be aware that using the same client attestation across multi
 
 The guidance provided by {{RFC7519}} and {{RFC8725}} applies.
 
-## Replay Attack Detection
+## Replay Attacks {#security-consideration-replay}
+
+An Authorization Server MUST implement measures to detect replay attacks by the Client Instance. In the context of this specification, this means to detect that an attacker is resending any Client Attestation PoP JWT an Authorization Server. The following options are RECOMMENDED for this client authentication method:
```

Comment:
> An Authorization Server MUST implement measures to detect replay attacks by the Client Instance.

This seems overly strong. I don't think we need replay protection requirements any stronger for this auth method than we have for DPoP or private_key_jwt do. In there IIRC it's a recommendation at best.

Reply:
Furthermore, it should be permitted to re-use the same challenge if new one was not received or manually retrieved.

Quoting from DPoP:

> Server-provided nonces are an effective means for further reducing the chances for successful DPoP proof replay. Unlike cryptographic nonces, it is acceptable for clients to use the same nonce multiple times and for the server to accept the same nonce multiple times. As long as the jti value is tracked and duplicates are rejected for the lifetime of the nonce, there is no additional risk of token replay.

What i'm looking for here is flexibility in the deployed protection proportional to the use case this method is deployed in.